### ISSUE TITLE
The bot is not functional after using `[p]backup` on non-JSON backends.

### ISSUE THREAD

**Issue by Jackenmen** [Cog-Creators/Red-DiscordBot#2959]
# Other bugs

#### What were you trying to do?

Do basically anything after doing a backup on PostgreSQL backend - for example run `[p]help`

#### What were you expecting to happen?

Bot to work fine.

#### What actually happened?

Basically nothing works before I restart.

#### How can we reproduce this issue?

Use `[p]backup` on a bot that uses PostgreSQL backend.
Try using `[p]help`

**Comment by Jackenmen** [comment]
A small update, this also happens on PostgreSQL backend.

**Comment by Jackenmen** [comment]
I don't really know what exactly happens during migration from non-JSON (i.e. PostgreSQL) to JSON backend (that migration happens before backup) but whatever it does, at the very least it makes all Config instances not aware of their default values (defaults are empty dict) which in result makes the whole bot not functional. After I tried to register global defaults for core conf, I was able to run `[p]help` just fine, but I can't say for sure that missing defaults are the only thing that is broken.
